salt/modules/keystore.py

```diff
@@ -112,18 +110,28 @@ def list(keystore, passphrase, alias=None, return_cert=False):
                     "Unsupported EntryType detected in keystore"
                 )
 
-            # Detect if ASN1 binary, otherwise assume PEM
-            if "\x30" in cert_result[0]:
-                public_cert = OpenSSL.crypto.load_certificate(ASN1, cert_result)
-            else:
-                public_cert = OpenSSL.crypto.load_certificate(PEM, cert_result)
-
+            public_cert = _get_cert(cert_result)
             entry_data.update(_parse_cert(entry_alias, public_cert, return_cert))
             decoded_certs.append(entry_data)
 
     return decoded_certs
 
 
+def _get_cert(certificate):
+    """
+    Gets the correct certificate depending of the encoding
+
+    :param certificate: str
+    """
+    ASN1 = OpenSSL.crypto.FILETYPE_ASN1
+    PEM = OpenSSL.crypto.FILETYPE_PEM
+    if certificate[0] == 0x30:
+        public_cert = OpenSSL.crypto.load_certificate(ASN1, certificate)
+    else:
+        public_cert = OpenSSL.crypto.load_certificate(PEM, certificate)
```

Comment:
**suggestion (nit-pick, non-blocking)** This *could* be

```suggestion
    if certificate[0] = 0x30:
        encoding = OpenSSL.crypto.FILETYPE_ASN1
    else:
        encoding = OpenSSL.crypto.FILETYPE_PEM
    public_cert =  OpenSSL.crypto.load_certificate(encoding, certificate)
```    

But... I don't think that's worth blocking the PR over. Good job :+1:

Reply:
Thanks @waynew.  If it's not blocking and it doesn't hurt your eyes, for now, I'll leave it as it is.